src/overlays/gamestates/ovl_daytelop/z_daytelop.c

```diff
@@ -1,22 +1,256 @@
+    s16 new_var;
+    u8 temp_v0_2;
+
+    this->transitionCountdown--;
+    if (this->transitionCountdown == 0) {
+        if (gSaveContext.day < 9) {
+            gSaveContext.gameMode = 0;
+        } else {
+            gSaveContext.unk_3F4A = 0xFFF6;
+            gSaveContext.day = 1;
+        }
+
+        // clang-format off
+        this->common.running = false; this->common.nextGameStateInit = Play_Init; this->common.nextGameStateSize = sizeof(GlobalContext);
+        // clang-format on
+        gSaveContext.time = CLOCK_TIME(6, 0);
+        D_801BDBC8 = 0xFE;
+    } else {
+        if (this->transitionCountdown == 90) {
+            this->fadeInState = DAYTELOP_HOURSTEXT_FADEIN;
+            this->alpha = 0;
+            D_80815FF0 = 30;
+        }
+        if (1) {}
+    }
+
+    temp_v0_2 = D_80815FF0;
+    new_var = this->fadeInState;
+    if (new_var == DAYTELOP_HOURSTEXT_FADEIN) {
+        s16 abs_temp = (s16)(ABS_ALT(this->alpha - 0xFF) / temp_v0_2);
+
+        this->alpha += abs_temp;
+        temp_v0_2--;
+        D_80815FF0 = temp_v0_2;
+
+        if (temp_v0_2 == 0) {
+            this->fadeInState = DAYTELOP_HOURSTEXT_ON;
+            this->alpha = 255;
+        }
+    }
+}
+#else
+void Daytelop_Update(DaytelopContext* this, GameState* gameState);
 #pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_daytelop/Daytelop_Update.s")
+#endif
+
+void Daytelop_Draw(DaytelopContext* this) {
+    GraphicsContext* gfxCtx = this->common.gfxCtx;
+
+    OPEN_DISPS(gfxCtx);
+    func_8012C628(this->common.gfxCtx);
+
+    if (gSaveContext.day >= 9) {
+        // Draw a white screen
+        gDPSetRenderMode(POLY_OPA_DISP++, G_RM_XLU_SURF, G_RM_XLU_SURF2);
+        gDPSetCombineMode(POLY_OPA_DISP++, G_CC_PRIMITIVE, G_CC_PRIMITIVE);
+        gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 255, 255, 255, 255);
+        gDPFillRectangle(POLY_OPA_DISP++, 0, 0, SCREEN_WIDTH, SCREEN_HEIGHT);
+    }
+
+    func_8012C628(this->common.gfxCtx);
+
+    gDPSetRenderMode(POLY_OPA_DISP++, G_RM_OPA_SURF, G_RM_OPA_SURF2);
+    gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 255, 255, 255, 255);
+
+    if (gSaveContext.day >= 9) {
+        gDPSetAlphaCompare(POLY_OPA_DISP++, G_AC_NONE);
+    }
+
+    // Draw the left side of the "Dawn of" texture
+    if (gSaveContext.day < 9) {
+        gDPLoadTextureBlock_4b(POLY_OPA_DISP++, sDayLeftTextures[CURRENT_DAY - 1], G_IM_FMT_I, 128, 64, 0,
+                               G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK,
+                               G_TX_NOLOD, G_TX_NOLOD);
+    } else {
+        gDPLoadTextureBlock_4b(POLY_OPA_DISP++, sDayLeftTextures[ARRAY_COUNT(sDayLeftTextures) - 1], G_IM_FMT_I, 128,
+                               64, 0, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK,
+                               G_TX_NOLOD, G_TX_NOLOD);
+    }
+    gSPTextureRectangle(POLY_OPA_DISP++, 0x0080, 0x0134, 0x0280, 0x0234, G_TX_RENDERTILE, 0, 0, 0x0400, 0x0400);
```

Comment:
```suggestion
    gSPTextureRectangle(POLY_OPA_DISP++, 32 << 2, 77 << 2, (32 + 128) << 2, (77 + 64) << 2, G_TX_RENDERTILE, 0, 0, 0x0400, 0x0400);
```
The 2nd, 3rd, 4th, 5th (ulx, uly, lrx, lry) arguments are for screen positioning. In this case, the upper-left position is (32, 77) and the screen dimensions are 128x64. The `<< 2` is for conversion to 10.2 fixed-point.